internal/plugin/service/serviceintegration/service_integration_data_source.go

```diff
@@ -0,0 +1,140 @@
+			"destination_endpoint_id": schema.StringAttribute{
+				Description: "Destination endpoint for the integration (if any)",
+				Computed:    true,
+				Validators:  []validator.String{endpointIDValidator},
+			},
+			"destination_service_name": schema.StringAttribute{
+				Description: "Destination service for the integration (if any)",
+				Required:    true,
+			},
+			"integration_type": schema.StringAttribute{
+				Description: "Type of the service integration. Possible values: " + schemautil.JoinQuoted(integrationTypes, ", ", "`"),
+				Required:    true,
+				Validators: []validator.String{
+					stringvalidator.OneOf(integrationTypes...),
+				},
+			},
+			"project": schema.StringAttribute{
+				Description: "Project the integration belongs to",
+				Required:    true,
+			},
+			"source_endpoint_id": schema.StringAttribute{
+				Description: "Source endpoint for the integration (if any)",
+				Computed:    true,
+				Validators:  []validator.String{endpointIDValidator},
+			},
+			"source_service_name": schema.StringAttribute{
+				Description: "Source service for the integration (if any)",
+				Required:    true,
+			},
+		},
+		Blocks: map[string]schema.Block{
+			"clickhouse_kafka_user_config":                clickhousekafka.NewDataSourceSchema(),
+			"clickhouse_postgresql_user_config":           clickhousepostgresql.NewDataSourceSchema(),
+			"datadog_user_config":                         datadog.NewDataSourceSchema(),
+			"external_aws_cloudwatch_metrics_user_config": externalawscloudwatchmetrics.NewDataSourceSchema(),
+			"kafka_connect_user_config":                   kafkaconnect.NewDataSourceSchema(),
+			"kafka_logs_user_config":                      kafkalogs.NewDataSourceSchema(),
+			"kafka_mirrormaker_user_config":               kafkamirrormaker.NewDataSourceSchema(),
+			"logs_user_config":                            logs.NewDataSourceSchema(),
+			"metrics_user_config":                         metrics.NewDataSourceSchema(),
+		},
+	}
+}
+
+// Read reads datasource
+// All functions adapted for resourceModel, so we use it as donor
+// Copies state from datasource to resource, then back, when things are done
+func (s *serviceIntegrationDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	var o dataSourceModel
+	resp.Diagnostics.Append(req.Config.Get(ctx, &o)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var res resourceModel
+	err := copier.Copy(&res, &o)
+	if err != nil {
+		resp.Diagnostics.AddError("data config copy error", err.Error())
+	}
+
+	dto, err := getSIByName(ctx, s.client, &res)
+	if err != nil {
+		resp.Diagnostics.AddError(errmsg.SummaryErrorReadingResource, err.Error())
+		return
+	}
+
+	loadFromDTO(ctx, &resp.Diagnostics, &res, dto)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	err = copier.Copy(&o, &res)
+	if err != nil {
+		resp.Diagnostics.AddError("dto copy error", err.Error())
+	}
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, o)...)
```

Comment:
use `util.ModelToPlanState`

Reply:
> use `util.ModelToPlanState`

`ModelToPlanState` does same thing without any extra profit. It returns bool value, that I don't need at all. I think `ModelToPlanState` is a bad idea.

Reply:
Cf. `organization.go` to understand the purpose of the boolean return value